### ISSUE TITLE
Error message only while typing on Ajax remote data loading

### ISSUE THREAD

**Issue by evilito** [select2/select2#4205]
## Steps to reproduce the issue

1. Create a select2 and configure it for ajax remote data loading.
2. Type your search string

## Expected behavior and actual behavior

When I follow those steps, I see "The results could not be loaded" as I type, and after I stop typing, I 1) get the actual results OR 2) get "No results found" if there was no match for my search.

I'm afraid this give the end users a sense of error, that something's wrong as they're typing. I understand that in order to be efficient, the ajax loading should not be searching until some miliseconds have passed from the last key press, but in this case it's better to not display anything that indicates an error, or just display "Searching..." as it was in previous versions.

I was expecting to see "Searching..." while I'm typing, and after I stop typing, 1) get the actual results OR 2) get "No results found" if there was no match for my search.

This bug also happens on the examples page ["Loading remote data section"](https://select2.github.io/examples.html#data-ajax). Just don't type too fast or you won't be able to see the error message.

## Environment

Browsers

- [x] Google Chrome
- [ ] Mozilla Firefox
- [ ] Internet Explorer

Operating System

- [x] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries

- jQuery version: 2.2.1
- Select2 version: v4.0.2-rc.1

## Isolating the problem

- [x] This bug happens [on the examples page](https://select2.github.io/examples.html)
- [x] The bug happens consistently across all tested browsers
- [x] This bug happens when using Select2 without other pluigns
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

**Comment by evilito** [comment]
IF this is intended, maybe it should be considered to create a new language string "onTyping" that says something that conveys the idea of "Waiting for you to finish your typing so I can actually perform a search".

**Comment by kevin-brown** [comment]
This has to do with us aborting in-progress requests when `delay` is in use. We should be ignoring errors that come from requests that have been aborted.

**Comment by XMadMax** [comment]
I still have the issue, also I can reproduce in the examples page.
Go to examples .... Loading remote data  ...
Try to type quick "aba aba aba ba aba aba PAUSEba ba ab PAUSEab aba PAUSEbbbabababababa"
the PAUSE means that you wait for response (No data found), and quickly type the next word.

**Comment by eric-pi** [comment]
I'm still seeing this issue, as well. The problem I'm seeing is in:

`if ($request.status && $request.status === '0')`

The value of the status property isn't a string, so the comparison in the second condition fails. But because it's a number, even if you remove the quotes around the 0 it still fails, since 0 && true evaluates to 0.

**Comment by orioljd** [comment]
this error still happens can someone reopen this? the way to prevent the show of the message can be:

if (request.statusText=='abort') // don't show the error message

I do it in other JS file to prevent a modal, if someone can modify the code

**Comment by lucas29252** [comment]
Thanks @orioljd, that works for me, but in some cases I also had to check first if the transport exposes a statusText property. So, finally the full condition is:

```javascript
if (($request.status && $request.status === '0') ||
    ($request.statusText && $request.statusText === 'abort')) {
  return;
}
```

**Comment by nelson6e65** [comment]
Is this solved? I'm still having this message when loading from ajax in slow connections.

**Comment by edariedl** [comment]
Hi, I just updated select2 to 4.0.5 and error message is still there when loading from ajax.

**Comment by JoshBleggi** [comment]
Same, also still seeing this issue

**Comment by alexweissman** [comment]
@edariedl @JoshBleggi you need to pull v4.0.6.

**Comment by vaindil** [comment]
I can confirm that the problem is resolved for me in 4.0.6-rc.1. (I didn't test rc.0.) Thank you!

**Comment by MichielDP** [comment]
Issue is not resolved for me in 4.0.6-rc.1. The request is never reaching my backend.
